core/templates/AppSpec.ts

```diff
@@ -92,23 +92,66 @@ describe('App', function() {
-        $exceptionHandler(error);
-        var expectedError = new Error(error);
-        expect(sourceMappedStackTrace.mapStackTrace).toHaveBeenCalledWith(
-          jasmine.stringMatching(regex),
-          jasmine.any(Function));
-        expect($log.error).toHaveBeenCalledWith(expectedError);
-      };
-      testException('something', /^Error: something/);
-      testException('', /^Error: /);
-      testException(undefined, /^Error: /);
-      testException(null, /^Error: null/);
-      testException({
-        a: 'something'
-      }, /^Error: \[object Object\]/);
-      testException({}, /^Error: \[object Object\]/);
+    it('should handle undefined gracefully', function() {
+      $exceptionHandler(undefined);
+      const errorFromObject = new Error('undefined');
+
+      expect(sourceMappedStackTrace.mapStackTrace).toHaveBeenCalledWith(
+        jasmine.stringMatching(/^Error: undefined/), jasmine.any(Function));
+      expect($log.error).toHaveBeenCalledWith(errorFromObject);
+    });
+
+    it('should handle null gracefully', function() {
+      $exceptionHandler(null);
+      const errorFromObject = new Error(null);
+
+      expect(sourceMappedStackTrace.mapStackTrace).toHaveBeenCalledWith(
+        jasmine.stringMatching(/^Error: null/), jasmine.any(Function));
+      expect($log.error).toHaveBeenCalledWith(errorFromObject);
+    });
+
+    it('should handle string values gracefully', function() {
+      $exceptionHandler('something');
+      const errorFromObject = new Error('something');
+
+      expect(sourceMappedStackTrace.mapStackTrace).toHaveBeenCalledWith(
+        jasmine.stringMatching(/^Error: something/), jasmine.any(Function));
+      expect($log.error).toHaveBeenCalledWith(errorFromObject);
+    });
+
+    it('should print stack trace when given empty string', function() {
+      $exceptionHandler('');
+      const errorFromObject = new Error('');
+
+      expect(sourceMappedStackTrace.mapStackTrace).toHaveBeenCalledWith(
+        jasmine.stringMatching(/^Error/), jasmine.any(Function));
+      expect($log.error).toHaveBeenCalledWith(errorFromObject);
+    });
+
+    it('should handle object values gracefully', function() {
+      // Error constructor will fail to compile without casting the object from
+      // unknown to string.
+      let obj: unknown = {a: 'something'};
+      $exceptionHandler(obj);
+      const errorFromObject = new Error(<string> obj);
+
+      expect(sourceMappedStackTrace.mapStackTrace).toHaveBeenCalledWith(
+        jasmine.stringMatching(/^Error: \[object Object\]/),
+        jasmine.any(Function));
+      expect($log.error).toHaveBeenCalledWith(errorFromObject);
+    });
+
+    it('should handle empty object values gracefully', function() {
+      // Error constructor will fail to compile without casting the object from
+      // unknown to string.
+      let obj: unknown = {};
+      $exceptionHandler(obj);
+      const errorFromObject = new Error(<string> obj);
```

Comment:
We usually do not put a space between the `<…>` and the variable.

```suggestion
      const errorFromObject = new Error(<string>obj);
```

Reply:
@brianrodri Btw, this error is also in other places where you introduced `<…>`

Reply:
Done